crates/rolldown/src/stages/link_stage/reference_needed_symbols.rs

```diff
@@ -98,8 +98,7 @@ impl LinkStage<'_> {
                       if matches!(
                         self.options.format,
                         OutputFormat::Cjs | OutputFormat::Iife | OutputFormat::Umd
-                      ) && !rec.meta.contains(ImportRecordMeta::IsPlainImport)
```

Comment:
This optimization is used to eliminate unused runtime code, see crates/rolldown/tests/rolldown/function/format/cjs/plain_import_should_not_introduce_to_esm/artifacts.snap as an example, but since now we generate different external import code like: 
```js
let import_xxx = require('external');
import_xxx = __toESM(import_xxx);
```
This optimization is no longer valid, reverted it.